File: api/src/main/java/com/vibe/pay/backend/payment/dto/PaymentInitiateRequest.java
package com.vibe.pay.backend.payment.dto;

import lombok.Getter;
import lombok.Setter;

/**
 * @author Claude
 * @version 1.0
 * @since 2025-01-28
 */
@Getter
@Setter
public class PaymentInitiateRequest {
    private Long memberId;
    private String orderId;
    private Double amount;
    private String paymentMethod;
    private String pgCompany;
    private Double usedMileage;
    private String goodName;
    private String buyerName;
    private String buyerTel;
    private String buyerEmail;
}

File: api/payment-service/src/main/kotlin/com/vibepay/payment/dto/PaymentInitiateRequest.kt
package com.vibepay.payment.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull

data class PaymentInitiateRequest(
    @field:NotBlank val orderNumber: String,
    @field:NotNull val amount: Long,
    val productName: String = "상품",
    val buyerName: String = "",
    val buyerTel: String = "",
    val buyerEmail: String = "",
    val paymentMethod: String = "Card"
)
